### PR TITLE
Layer update strategies

### PR DESCRIPTION
**Pull request by peppsac**
Implement different (and configurable) update strategies for layer.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -34,321 +34,331 @@
...
+                }
+            });
 
-                itowns.viewer.addImageryLayer({
+            itowns.viewer.addImageryLayer({
                 protocol:   "wmts",
                 id:         "ScanEX",
                 url:        "http://wxs.ign.fr/va5orxd0pgzvq3jxutqfuy0b/geoportail/wmts",
                 fx :        2.5,
+                updateStrategy: {
```
Comment (gchoqueux) [inline index.html:272]: should be added default value when there's no updateStrategy

**Review by gchoqueux — CHANGES_REQUESTED** [review]

**Comment by gchoqueux** [comment]
No updating/reloading when a layer going from invisible to visible

**Comment by nosy-b** [comment]
Same bug as the master:
When going **outside of France** even with only 'Ortho' imagery layer activated it tries to download other layers data deactivated(ex: SCAN-EXPRESS.STANDARD)

**Comment by peppsac** [comment]
> Same bug as the master:
> When going outside of France even with only 'Ortho' imagery layer activated it tries to download other layers data deactivated(ex: SCAN-EXPRESS.STANDARD)

It's expected.
To be `loaded` a tile needs to have all of its textures, regardless of the status of the layer.
So while I agree with you that this is _wrong_, a proper fix would be to rework the `child.texturesNeeded` mechanism.

Note : a tile expects to have (at least) 1 texture per layer where `tileInsideLimit` returns true.

File: index.html (hunk 1 above)
Comment (peppsac) [reply index.html:272]: Fixed.

**Comment by peppsac** [comment]
> No updating/reloading when a layer going from invisible to visible

Fixed as well (calling `update()` instead of `scene.renderScene3D()` in `ApiGlobe.prototype.setLayerVisibility`)

**Comment by gchoqueux** [comment]
r+

**Pull request merged**